pkgs/misc/drivers/xp-pen-deco-01-v2-driver/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib, stdenv, fetchzip, libusb, glibc, libGL, xorg, qtx11extras, wrapQtAppsHook, autoPatchelfHook, ... }:
+
+stdenv.mkDerivation {
+  pname = "xp-pen-deco-01-v2-driver";
+  version = "3.2.3.220323-1";
+  src = fetchzip {
+    url = "https://www.xp-pen.com/download/file/id/1936/pid/440/ext/gz.html#.tar.gz";
+    sha256 = "sha256-n/yutkRsjcIRRhB4q1yqEmaa03/1SO8RigJi/ZkfLbk=";
+  };
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    libusb
+    xorg.libX11
+    xorg.libXtst
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXinerama
+    glibc
+    libGL
+    stdenv.cc.cc.lib
+    qtx11extras
+  ];
+
+  installPhase = let dataDir = "var/lib/xppend1v2"; in ''
+    runHook preInstall
+
+    mkdir -p $out/{opt,bin}
+    cp -r App/usr/lib/pentablet/{pentablet,resource.rcc,conf} $out/opt
+    chmod +x $out/opt/pentablet
+    cp -r App/lib $out/lib
+    sed -i 's#usr/lib/pentablet#${dataDir}#g' $out/opt/pentablet
+    makeWrapper $out/opt/pentablet $out/bin/xp-pen-deco-01-v2-driver \
+      --run 'if [ "$EUID" -ne 0 ]; then echo "Please run as root."; exit 1; fi' \
+      --run 'if [ ! -d /${dataDir} ]; then mkdir -p /${dataDir}; cp -r '$out'/opt/conf /${dataDir}; chmod u+w -R /${dataDir}; fi'
```

Comment:
Does the binary run at all without it? If i recall correctly the existing `pentablet-driver` package does.

In my opinion this should be apart of a nixos module instead, that would take care of removing the file if the user uninstalls the driver, and updating when we update this package.

Reply:
Without the configuration, the binary does in fact launch, but it is completely nonfunctional (it gets stuck on the "Please connect device" screen). I'm not entirely sure that this should be a NixOS module, since the configuration includes stuff like 2,766 lines of INI that specify all of the different translations of every text in the UI.

Reply:
> Without the configuration, the binary does in fact launch, but it is completely nonfunctional (it gets stuck on the "Please connect device" screen).

Ah, alright 👍

> I'm not entirely sure that this should be a NixOS module, since the configuration includes stuff like 2,766 lines of INI that specify all of the different translations of every text in the UI.

The nixos module could also just install the file from this packages output to `/var/...`, instead of exposing configuration options.